stan/math/prim/scal/fun/F32.hpp

```diff
@@ -1,42 +1,89 @@
 #ifndef STAN_MATH_PRIM_SCAL_FUN_F32_HPP
 #define STAN_MATH_PRIM_SCAL_FUN_F32_HPP
 
+#include <stan/math/prim/scal/fun/sign.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <stan/math/prim/scal/fun/is_inf.hpp>
 #include <cmath>
 
 namespace stan {
   namespace math {
 
+    /**
+     * Hypergeometric function, 3F2.
+     *
+     * The generalized hypergeometric function is a power series. This
+     * implementation computes the power series directly stopping when
+     * the series converges to within <code>precision</code> or takes
+     * <code>max_steps</code>.
+     *
+     * Although some convergence conditions and divergent conditions are known,
+     * this function does not check the inputs for known convergent conditions.
+     *
+     * This function will converge if:
+     *   - <code>a1</code>, <code>a2</code>, or <code>a3</code> is a
+     *     non-positive integer
+     *   - <code>b1</code> or <code>b2</code> is a non-positive integer
+     *   - <code>z</code> is less than 1
+     * This function will diverge if <code>z</code> is greater than 1.
+     * When <code>z</code> is 1, which is the case for the beta binomial
+     * cdf, it is hard to determine.
```

Comment:
👍 , I haven't fixed all the doc yet, I have a reference for when it does converge.  Do you think we should be checking for convergence conditions at the front? The original function didn't do this and I was aiming for more of a bugfix here with the intention of improving the functions as a second PR.  I'll either include the improvements here or make a specific second PR.